extract.js

```diff
@@ -9,53 +9,57 @@ const pipe = BB.promisify(require('mississippi').pipe)
 const optCheck = require('./lib/util/opt-check')
 const retry = require('promise-retry')
 const rimraf = BB.promisify(require('rimraf'))
+const readdirAsync = BB.promisify(require('fs').readdir)
 
 module.exports = extract
 function extract (spec, dest, opts) {
   opts = optCheck(opts)
   spec = typeof spec === 'string' ? npa(spec, opts.where) : spec
   const startTime = Date.now()
-  if (opts.integrity && opts.cache && !opts.preferOnline) {
-    opts.log.silly('pacote', 'trying', spec.name, 'by hash:', opts.integrity.toString())
-    return extractByDigest(
-      startTime, spec, dest, opts
-    ).catch(err => {
-      if (err.code === 'ENOENT') {
-        opts.log.silly('pacote', `data for ${opts.integrity} not present. Using manifest.`)
-        return extractByManifest(startTime, spec, dest, opts)
-      }
+  checkOverwrite(opts.extractOverwrite, spec, dest)
```

Comment:
You should add a `return` here, so that this promise and its sub-promises become the return value for `extract`.